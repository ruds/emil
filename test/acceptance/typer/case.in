case (23, "hi there")
 of (23, s) => s
  | (_, "hi there") => "bye"
  | _ => "huh";
case [1, 2, 3]
 of nil => "empty"
  | (::) (_, nil) => "one"
  | (::) (_, _) => "many";
case {a = (1, "hi"), b = 2.3, c = 12}
 of {a, c = 1, ...} => a
  | {c = 2, ...} => (2, "bye")
  | {a, ...} => a;
case {a = (1, "hi"), b = 2.3, c = 12}
 of {a, c = 1, ...} => a
  | {c = 2, ...} => (2, "bye");
case {a = (1, "hi"), b = 2.3, c = 12}
 of {a, c = 1, ...} => a
  | {c = 2, ...} => (2, "bye")
  | {a, ...} => a
  | {b, ...} => (3, "redundant");
case 1 of "hi" => 2;
