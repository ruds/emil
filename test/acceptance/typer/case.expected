@0001:
val it : string
@0005:
val it : string
@0009:
val it : (bigint * string)
@0013: WARNING
@0013:
val it : (bigint * string)
@0016: ERROR
@0021: WARNING
@0021: ERROR
@0022:
